File: backend/src/main/java/tn/itbs/backend/entites/Consultation.java
package tn.itbs.backend.entites;


import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data

@NoArgsConstructor
@AllArgsConstructor
public class Consultation {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int idConsultation ;
	@JoinColumn(name = "idRendezVous")
    private RendezVous rendezVous;
    @Column(nullable = false)
    private String diagnostic;
    @Column(nullable = false)
    private String ordonnance;
    @Column(nullable = false)
    private Float prix;
	
}

File: frontend/frontend/src/app/features/consultation/models/consultation.ts
export interface Consultation {
  idConsultation: number;
  diagnostic: string;
  ordonnance: string;
  prix: number;
  dateConsultation?: string;
  idRendezVous: number;
}

export interface ConsultationDTO {
  idConsultation: number;
  diagnostic: string;
  ordonnance: string;
  prix: number;
  idRendezVous: number;
}
